indico_toolkit/indico_wrapper/modelop.py

```diff
@@ -0,0 +1,200 @@
+                        optimize_for : "predict_speed" ( "predict_speed", "accuracy", "speed", "accuracy_fp16" and "predict_speed_fp16")
+                        subtoken_predictions : True
+                        base_model : "roberta" ("roberta", "small" (distilled version of RoBERTa), "multilingual", "fast", "textcnn", "fasttextcnn")
+                        class_weight : "sqrt" ("linear", "sqrt", "log", None)
+
+                    For Text Classification Model:
+                        model_type : "standard" (“tfidf_lr”, “tfidf_gbt”, “standard”, “finetune”)
+        Returns:
+            dict: Dictionary of advanced model training options
+        """
+        model = self.client.call(
+            GraphQLRequest(
+                """
+                mutation updateModelGroup($modelGroupId: Int!, $modelTrainingOptions: JSONString) {
+                    updateModelGroupSettings(
+                      modelGroupId: $modelGroupId
+                      modelTrainingOptions: $modelTrainingOptions
+                  ) {
+                      modelOptions {
+                          id
+                          modelTrainingOptions
+                        }
+                    }
+                }
+                """,
+                {
+                    "modelGroupId": model_group_id,
+                    "modelTrainingOptions": json.dumps(
+                        self._parameter_check(model_type, **kwargs)
+                    ),
+                },
+            )
+        )
+        options = json.loads(
+            model["updateModelGroupSettings"]["modelOptions"]["modelTrainingOptions"]
+        )
+        options["id"] = model["updateModelGroupSettings"]["modelOptions"]["id"]
+        return options
+
+    def _parameter_check(self, model_type, **kwargs):
+        text_extraction_params = {
+            "max_empty_chunk_ratio": lambda value: 0 <= value <= 1.0e5,
+            "auto_negative_scaling": [True, False],
+            "optimize_for": [
+                "predict_speed",
+                "accuracy",
+                "speed",
+                "accuracy_fp16",
+                "predict_speed_fp16",
+            ],
+            "subtoken_predictions": [True, False],
+            "base_model": [
+                "roberta",
+                "small",
+                "multilingual",
+                "fast",
+                "textcnn",
+                "fasttextcnn",
+            ],
+            "class_weight": ["linear", "sqrt", "log", None],
+        }
+
+        text_classification_params = {
+            "model_type": ["tfidf_lr", "tfidf_gbt", "standard", "finetune"]
+        }
+
+        if model_type == "text_extraction":
+            acceptable_params = text_extraction_params
+        elif model_type == "text_classification":
+            acceptable_params = text_classification_params
+        else:
+            raise ValueError(f"Invalid model type: {model_type}")
+
+        model_params = {}
```

Comment:
I might be missing something, but I believe model_params will be an exact copy of kwargs or the method will `raise`. In which case you can simplify it quite a bit by doing just `if invalid: raise` checks and return kwargs at the end otherwise.